src/pages/index.js

```diff
@@ -0,0 +1,307 @@
+const deleteFormElement = document.querySelector("#delete-form");
+
+// Preview image popup elements
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image")
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption")
+const previewModalCloseButton = previewModal.querySelector(".modal__close-button-preview");
+
+const closeButtons = document.querySelectorAll(".modal__close");
+
+function handleLike(evt, id, cardLikeButton) {
+  const isLiked = cardLikeButton.classList.contains("card__like-button_liked");
+
+  api
+    .changeLikeStatus(id, isLiked)
+    .then(() => {
+      cardLikeButton.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+
+previewModalCloseButton.addEventListener("click", () => {
+  closeModal(previewModal);
+})
+
+let selectedCard, selectedCardId;
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", modalEscClose);
+  modal.addEventListener("mousedown", modalOverlayClose);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", modalEscClose);
+  modal.removeEventListener("mousedown", modalOverlayClose);
+}
+
+function modalEscClose(evt) {
+  if (evt.key === "Escape") {
+    const modal = document.querySelector(".modal_opened");
+    closeModal(modal);
+  }
+}
+
+function modalOverlayClose(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+
+  setButtonText(submitButton, true, "Save", "Saving...");
+
+  const updatedUserInfo = {
+    name: editModalNameInput.value,
+    about: editModalDescriptionInput.value,
+  };
+
+  api
+    .editUserInfo(updatedUserInfo)
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
+     // setButtonText(cardSubmitButton, false);
+    });
+} 
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  resetValidation(editProfileFormElement, [editModalNameInput, editModalDescriptionInput]
+    , settings);
+  openModal(editModal);
+});
+
+closeButtons.forEach((button) => {
+  const popup = button.closest(".modal");
+  button.addEventListener('click', () => closeModal(popup));
+});
+
+cardModalButton.addEventListener("click", () => {
+  openModal(cardModal);
+});
+
+  editProfileFormElement.addEventListener("submit", handleEditFormSubmit);
+  cardFormElement.addEventListener("submit", handleAddCardSubmit);
+
+avatarModalButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
```

Comment:
You need to **format** the markup (and code) to show **indentation** and to make the code more readable. You can do that with `Prettier` or a combination `SHIFT + ALT + F`